pkgs/hooks_runner/lib/src/build_runner/build_runner.dart

```diff
@@ -204,23 +208,24 @@ class NativeAssetsBuildRunner {
   ///
   /// The base protocol can be extended with [extensions]. See
   /// [ProtocolExtension] for more documentation.
-  Future<LinkResult?> link({
+  Future<Result<LinkResult, HooksRunnerFailure>> link({
     required List<ProtocolExtension> extensions,
     Uri? resourceIdentifiers,
     required BuildResult buildResult,
   }) async {
     final loadedUserDefines = await _loadedUserDefines;
     final hookResultUserDefines = await _checkUserDefines(loadedUserDefines);
-    if (hookResultUserDefines == null) {
-      return null;
+    if (hookResultUserDefines.isFailure) {
+      return hookResultUserDefines;
```

Comment:
related: here the `asFailure` is missing and it seems to be fine?

Reply:
> related: here the `asFailure` is missing and it seems to be fine?

The `T` in `Result` for this method and `hookResultUserDefines` is identical, so we don't need the `Never` in for the `T` in `Failure`.